sdk/eventgrid/eventgrid/src/generated/models/index.ts

```diff
@@ -10,3357 +10,2141 @@ import * as coreHttp from "@azure/core-http";
    */
   readonly state?: MediaJobState;
-  /**
-   * Gets the Job correlation data.
-   */
+  /** Gets the Job correlation data. */
   correlationData?: { [propertyName: string]: string };
 }
 
-/**
- * Details of JobOutput errors.
- */
+/** Details of JobOutput errors. */
 export interface MediaJobError {
   /**
    * Error code describing the error.
+   * NOTE: This property will not be serialized. It can only be populated by the server.
    */
   readonly code?: MediaJobErrorCode;
   /**
    * A human-readable language-dependent representation of the error.
+   * NOTE: This property will not be serialized. It can only be populated by the server.
    */
   readonly message?: string;
   /**
    * Helps with categorization of errors.
+   * NOTE: This property will not be serialized. It can only be populated by the server.
    */
   readonly category?: MediaJobErrorCategory;
   /**
    * Indicates that it may be possible to retry the Job. If retry is unsuccessful, please contact Azure support via Azure Portal.
+   * NOTE: This property will not be serialized. It can only be populated by the server.
    */
   readonly retry?: MediaJobRetry;
   /**
    * An array of details about specific errors that led to this reported error.
+   * NOTE: This property will not be serialized. It can only be populated by the server.
    */
   readonly details?: MediaJobErrorDetail[];
 }
 
-/**
- * Details of JobOutput errors.
- */
+/** Details of JobOutput errors. */
 export interface MediaJobErrorDetail {
   /**
    * Code describing the error detail.
+   * NOTE: This property will not be serialized. It can only be populated by the server.
    */
   readonly code?: string;
   /**
    * A human-readable representation of the error.
+   * NOTE: This property will not be serialized. It can only be populated by the server.
    */
   readonly message?: string;
 }
 
-/**
- * The event data for a Job output.
- */
+/** The event data for a Job output. */
 export interface MediaJobOutput {
-  /**
-   * Polymorphic discriminator, which specifies the different types this object can be
-   */
-  "@odata.type": "#Microsoft.Media.JobOutputAsset";
-  /**
-   * The discriminator for derived types.
-   */
-  odataType?: string;
-  /**
-   * Gets the Job output error.
-   */
+  /** Polymorphic discriminator, which specifies the different types this object can be */
+  odataType: "#Microsoft.Media.JobOutputAsset";
+  /** Gets the Job output error. */
   error?: MediaJobError;
-  /**
-   * Gets the Job output label.
-   */
+  /** Gets the Job output label. */
   label?: string;
-  /**
-   * Gets the Job output progress.
-   */
+  /** Gets the Job output progress. */
   progress: number;
-  /**
-   * Gets the Job output state.
-   */
+  /** Gets the Job output state. */
   state: MediaJobState;
 }
 
-/**
- * The event data for a Job output asset.
- */
-export type MediaJobOutputAsset = MediaJobOutput & {
```

Comment:
Were there changes in the SWAGGER? I see this type has a new property odataType (below). Also saw a couple of other definitions with new properties

Reply:
No. These changes are due to the generator. In old generator, when we have the odata type only the base class has odata type defined (possibly with a union type). With the new generator, even the subclasses have their own odata types and the base class has its own odata type (mostly not used at all)